src/v/kafka/server/group.cc

```diff
@@ -1447,6 +1447,96 @@ ss::future<error_code> group::remove() {
     co_return error_code::none;
 }
 
+ss::future<>
+group::remove_topic_partitions(const std::vector<model::topic_partition>& tps) {
+    std::vector<std::pair<model::topic_partition, offset_metadata>> removed;
+    for (const auto& tp : tps) {
+        _pending_offset_commits.erase(tp);
+        if (auto offset = _offsets.extract(tp); offset) {
+            removed.emplace_back(
+              std::move(offset.key()), std::move(offset.mapped()));
+        }
+    }
+
+    // if no members and no offsets
+    if (
+      in_state(group_state::empty) && _pending_offset_commits.empty()
+      && _offsets.empty()) {
+        vlog(
+          klog.debug,
+          "Marking group {} as dead at {} generation",
+          _id,
+          generation());
+
+        set_state(group_state::dead);
+    }
+
+    if (removed.empty()) {
+        co_return;
+    }
+
+    _offsets.rehash(0);
```

Comment:
right. maybe we can look at a the template type or inner types of the maps to set some specific load factors. maybe we always set it to 100% capacity. so it does this automagically?

Reply:
can you create a ticket? this would presumably be something that applies to the entire code base

Reply:
sure, i created #622